internal/dev_server/dev_server.go

```diff
@@ -0,0 +1,93 @@
+package dev_server
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/adrg/xdg"
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+	"github.com/launchdarkly/ldcli/internal/client"
+	"github.com/launchdarkly/ldcli/internal/dev_server/adapters"
+	"github.com/launchdarkly/ldcli/internal/dev_server/api"
+	"github.com/launchdarkly/ldcli/internal/dev_server/db"
+	"github.com/launchdarkly/ldcli/internal/dev_server/model"
+	"github.com/launchdarkly/ldcli/internal/dev_server/sdk"
+	"github.com/launchdarkly/ldcli/internal/dev_server/ui"
+)
+
+type Client interface {
+	RunServer(ctx context.Context, serverParams ServerParams)
+}
+
+type ServerParams struct {
+	AccessToken  string
+	BaseURI      string
+	DevStreamURI string
+	Port         string
+}
+
+type LDClient struct {
+	cliVersion string
+}
+
+var _ Client = LDClient{}
+
+func NewClient(cliVersion string) LDClient {
+	return LDClient{cliVersion: cliVersion}
+}
+
+func (c LDClient) RunServer(ctx context.Context, serverParams ServerParams) {
+	ldClient := client.New(serverParams.AccessToken, serverParams.BaseURI, c.cliVersion)
+	dbPath := getDBPath()
+	log.Printf("Using database at %s", dbPath)
+	sqlStore, err := db.NewSqlite(ctx, getDBPath())
+	if err != nil {
+		log.Fatal(err)
+	}
+	ss := api.NewStrictServer()
+	apiServer := api.NewStrictHandlerWithOptions(ss, nil, api.StrictHTTPServerOptions{
+		RequestErrorHandlerFunc:  RequestErrorHandler,
+		ResponseErrorHandlerFunc: ResponseErrorHandler,
+	})
+	r := mux.NewRouter()
+	r.Use(adapters.Middleware(*ldClient, serverParams.DevStreamURI))
+	r.Use(model.StoreMiddleware(sqlStore))
+	r.Use(model.ObserversMiddleware(model.NewObservers()))
+	r.Handle("/", http.RedirectHandler("/ui/", http.StatusFound))
+	r.Handle("/ui", http.RedirectHandler("/ui/", http.StatusMovedPermanently))
+	r.PathPrefix("/ui/").Handler(http.StripPrefix("/ui/", ui.AssetHandler))
+	sdk.BindRoutes(r)
+	handler := api.HandlerFromMux(apiServer, r)
+	handler = handlers.CombinedLoggingHandler(os.Stdout, handler)
+	handler = handlers.RecoveryHandler(handlers.PrintRecoveryStack(true))(handler)
+
+	addr := fmt.Sprintf("0.0.0.0:%s", serverParams.Port)
+	log.Printf("Server running on %s", addr)
+	log.Printf("Access the UI for toggling overrides at http://localhost:%s/ui or by running `ldcli dev-server ui`", serverParams.Port)
+	server := http.Server{
+		Addr:    addr,
+		Handler: handler,
+	}
+	log.Fatal(server.ListenAndServe())
+}
+
+func ResponseErrorHandler(w http.ResponseWriter, r *http.Request, err error) {
+	log.Printf("Error while serving response: %+v", err)
+	http.Error(w, err.Error(), http.StatusInternalServerError)
+}
+func RequestErrorHandler(w http.ResponseWriter, r *http.Request, err error) {
+	log.Printf("Error while serving request: %+v", err)
+	http.Error(w, err.Error(), http.StatusBadRequest)
+}
+
+func getDBPath() string {
+	dbFilePath, err := xdg.StateFile("ldcli/dev_server.db")
```

Comment:
DB is meant to be persistent, but it should be exclusively owned by a single server instance (if two try and start, one will crash because it won't be able to bind a port). I'm, TBH, not a SQLite expert. Do we still need that parameter in this scenario?